workers/social/lib/social/models/group/index.coffee

```diff
@@ -1493,65 +1493,106 @@ module.exports = class JGroup extends Module
       if err then callback err
       else oldAddOwner.call this, target, options, callback
 
+
+  checkUserPassword: (account, password, callback) ->
+
+    unless account or password
+      return callback 'Error occured while deleting team'
+
+    account.fetchUser (err, user) ->
+      unless err
+        unless user.checkPassword password
+          return callback 'Your password didn\'t match with our records'
+        else
+          callback()
+
+
   destroy    : permit
     advanced : [
       { permission: 'edit own groups', validateWith : Validators.own }
       { permission: 'edit groups',     superadmin   : yes }
     ]
-    success  : (client, callback) ->
+    success  : (client, password, callback) ->
 
-      JName = require '../name'
+      account = client?.connection?.delegate
+      { sessionToken } = client
 
-      removeHelper = (model, err, next) ->
-        return next err  if err
-        return next()  unless model
+      # mark this team as deleted
+      @update { $set: { markAs: 'deleted' } }, (err) ->
```

Comment:
Mark this team as deleted to specify that this team should be deleted by worker if this operations are failed??